src/model/types/account_id/eip155.rs

```diff
@@ -0,0 +1,155 @@
+}
+
+pub fn validate_eip155(reference: &str, address: &str) -> Result<(), Eip155Error> {
+    validate_eip155_reference(reference)?;
+    validate_eip155_address(address)?;
+    Ok(())
+}
+
+#[derive(Debug, PartialEq, Eq, Error)]
+pub enum Eip155ReferenceError {
+    #[error("does not match regex")]
+    Regex,
+}
+
+fn validate_eip155_reference(reference: &str) -> Result<(), Eip155ReferenceError> {
+    if REFERENCE_PATTERN.is_match(reference) {
+        Ok(())
+    } else {
+        Err(Eip155ReferenceError::Regex)
+    }
+}
+
+#[derive(Debug, PartialEq, Eq, Error)]
+pub enum Eip155AddressError {
+    #[error("does not match regex")]
+    Regex,
+
+    #[error("does not pass ERC-55 checksum")]
+    Erc55,
+}
+
+fn validate_eip155_address(address: &str) -> Result<(), Eip155AddressError> {
+    if let Some(caps) = ADDRESS_PATTERN.captures(address) {
+        let (_, [address_hex]) = caps.extract();
+        let erc55: String =
+            erc_55_checksum_encode(&address_hex.to_ascii_lowercase()).collect::<String>();
+        if erc55 == address_hex {
+            Ok(())
+        } else {
+            Err(Eip155AddressError::Erc55)
+        }
+    } else {
+        Err(Eip155AddressError::Regex)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use {super::*, crate::model::types::account_id::caip10::validate_caip_10};
+
+    #[test]
+    fn test_eip155() {
+        // https://github.com/ChainAgnostic/namespaces/blob/main/eip155/caip10.md#test-cases
+
+        // Ethereum mainnet (valid/checksummed)
+        let test = "eip155:1:0x22227A31dd842196A246d8f3b775998560eAa61d";
+        assert!(validate_caip_10(test).is_ok());
+
+        // Ethereum mainnet (will not validate in EIP155-conformant systems)
+        let test = "eip155:1:0x22227a31dd842196a246d8f3b775998560eaa61d";
+        assert!(validate_caip_10(test).is_err());
+
+        // Polygon mainnet (valid/checksummed)
+        let test = "eip155:137:0x0495766cD136138Fc492Dd499B8DC87A92D6685b";
+        assert!(validate_caip_10(test).is_ok());
+
+        // Polygon mainnet (will not validate in EIP155-conformant systems)
+        let test = "eip155:137:0x0495766CD136138FC492DD499B8DC87A92D6685B";
+        assert!(validate_caip_10(test).is_err());
+    }
+
+    #[test]
+    fn validate_fn_uses_address_check() {
+        let address = "123";
+        let reference = "1";
+        assert_eq!(
+            validate_eip155(reference, address),
+            Err(Eip155Error::Address(Eip155AddressError::Regex))
+        );
+    }
+
+    #[test]
+    fn validate_fn_uses_reference_check() {
+        let address = "0x62639418051006514eD5Bb5B20aa7aAD642cC2d0";
+        let reference = "abc";
+        assert_eq!(
+            validate_eip155(reference, address),
+            Err(Eip155Error::Reference(Eip155ReferenceError::Regex))
+        );
+    }
+
+    #[test]
+    fn test_is_valid_eip155_account() {
+        assert!(validate_caip_10("eip155:1:0x62639418051006514eD5Bb5B20aa7aAD642cC2d0").is_ok());
```

Comment:
consider turning these strings into a constant

Reply:
I don't think a constant would make these test cases easier to understand